phantazm-neuron/src/main/java/com/github/phantazmnetwork/neuron/navigator/GroundNavigator.java

```diff
@@ -0,0 +1,241 @@
+package com.github.phantazmnetwork.neuron.navigator;
+
+import com.github.phantazmnetwork.commons.vector.Vec3D;
+import com.github.phantazmnetwork.commons.vector.Vec3I;
+import com.github.phantazmnetwork.neuron.agent.Agent;
+import com.github.phantazmnetwork.neuron.engine.PathEngine;
+import com.github.phantazmnetwork.neuron.node.Node;
+import com.github.phantazmnetwork.neuron.operation.PathResult;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+import java.util.function.ToIntFunction;
+
+/**
+ * Standard {@link TrackingNavigator} implementation for ground-based movement.
+ */
+public class GroundNavigator extends TrackingNavigator {
+    private static final double NODE_REACHED_DISTANCE = 0.25;
+
+    private final long immobileThreshold;
+    private final long missingStartDelay;
+    private final ToIntFunction<? super PathResult> explorationDelayFunction;
```

Comment:
not that it should reasonably change anything, but why not ToLongFunction for consistency?